routes/channel.js

```diff
@@ -48,6 +51,25 @@ function getStatus(withTree, req, res) {
 	})
 }
 
+function getEvents(req, res, next){
+	const { uid } = req.session
+	const resp = { channel: req.channel }
+
+	const eventsCol = db.getMongo().collection('eventAggregates')
+	const key = `events.IMPRESSION.${uid}`
+
+	return eventsCol
+	.find({
+		[ key ]:  { "$exists": true }
+	})
```

Comment:
you should project for the key here, in order to avoid leaking other publisher's data

Reply:
The other publisher data present would be the publisher balance and its exposed already on the blockchain

Reply:
okay done